client/dashboard/sites/settings-repositories/disconnect-repository-modal-content.tsx

```diff
@@ -75,14 +105,14 @@ export function DisconnectRepositoryModal( {
 					) }
 				</Text>
 
-				<ToggleControl
-					__nextHasNoMarginBottom
-					label={ __( 'Remove associated files from my WordPress.com site' ) }
-					checked={ removeFilesChecked }
-					onChange={ () => setRemoveFilesChecked( ( value ) => ! value ) }
+				<DataForm< { enabled: boolean } >
+					data={ data }
+					fields={ fields }
+					form={ form }
+					onChange={ handleRemoveFilesChange }
```

Comment:
For this simple component, maybe `data` and `onChange` can be inlined instead of calling the external fields [here](https://github.com/Automattic/wp-calypso/pull/105749/commits/ea3b07d1cc103529050f32f427abf9a26ee5ff0d#diff-34358b0ec6503f559b95426703e856b819757bd48d7ccea5de66c71072f97766R79) and [here](https://github.com/Automattic/wp-calypso/pull/105749/commits/ea3b07d1cc103529050f32f427abf9a26ee5ff0d#diff-34358b0ec6503f559b95426703e856b819757bd48d7ccea5de66c71072f97766R50), but I have left it like this for consistency with other examples

Reply:
Would it be better as follows?

```js
const [ data, setFormData ] = useState( { removeFilesChecked: false } );

...
onChange={ ( edits: Partial< FormData > ) => {
	setFormData( ( data ) => ( { ...data, ...edits } ) );
} }

...
deleteDeployment( data.removeFilesChecked, ... )
```

Reply:
It might seem more complex at first, but it is a generic pattern for all forms, so the consistency is a plus here, changed it as part of [3bb278f](https://github.com/Automattic/wp-calypso/pull/105749/commits/3bb278fbd77acd8e889d034fbea6d37c9dbdef69)